src/Plugin/KeyType/OauthKeyType.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Plugin\KeyType;
+
+use Apigee\Edge\HttpClient\Plugin\Authentication\Oauth;
+use Drupal\apigee_edge\OauthTokenStorage;
+use Drupal\apigee_edge\Plugin\EdgeOAuthKeyTypeInterface;
+use Drupal\key\KeyInterface;
+use Http\Message\Authentication;
+
+/**
+ * Key type for Apigee Edge basic authentication credentials.
+ *
+ * @KeyType(
```

Comment:
Can't you subclass this `KeyType` annotation class as `EdgeKeyType`, where could set some default settings for our keys like 
* group => "apigee_edge"
* multivalue=>TRUE
* origanization, endpoint, username are required", etc

and use `@EdgeKeyType` in the  annotation block? That way you would not need to repeat yourself here.